packages/babel-core/src/config/helpers/deep-array.ts

```diff
@@ -10,7 +10,9 @@ export function finalize<T>(deepArr: DeepArray<T>): ReadonlyDeepArray<T> {
   return Object.freeze(deepArr) as ReadonlyDeepArray<T>;
 }
 
-export function flattenToSet<T>(arr: ReadonlyDeepArray<T>): Set<T> {
+export function flattenToSet<T extends string>(
```

Comment:
`T` is constrained to `string` so that `Array.isArray(el)` can effectively differentiate between `ReadonlyDeepArray<T>` and `T`.